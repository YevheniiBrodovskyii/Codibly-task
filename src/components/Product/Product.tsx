import { useState, FC } from 'react'
import { IProduct } from './product.interface'

import ProductModal from '../ProductModal/ProductModal'

import {
	ProductContainer,
	ProductId,
	ProductName,
	ProductYear,
	ProductSpan,
} from './styled'

const Product: FC<{ product: IProduct }> = ({ product }) => {
	const [mounted, setMounted] = useState<boolean>(false)
	const [modal, setModal] = useState<boolean>(false)

	return (
		<>
			<ProductContainer
				color={product.color}
				onMouseEnter={() => setMounted(true)}
				onMouseLeave={() => setMounted(false)}
				onClick={() => setModal(true)}
			>
				<ProductId>Item id: {product.id}</ProductId>
				<ProductName>{product.name}</ProductName>
				<ProductSpan mounted={mounted}>Click for more...</ProductSpan>
				<ProductYear>{product.year}</ProductYear>
			</ProductContainer>
			{modal ? (
				<ProductModal product={product} modal={modal} setModal={setModal} />
			) : null}
		</>
	)
}

export default Product
