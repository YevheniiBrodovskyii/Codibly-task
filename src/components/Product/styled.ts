import styled from 'styled-components'

export const ProductContainer = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	width: 280px;
	height: 280px;
	border-radius: 10px;
	background-color: ${({ color }) => (color ? color : 'black')};
	margin: 0 auto;
	padding: 10px;
	transition: 0.3s all;
	cursor: pointer;
	&:hover {
		transform: scale(1.1);
	}
`

export const ProductId = styled.h3`
	width: 120px;
	text-align: center;
	color: white;
	border-radius: 10px;
	padding: 5px;
	margin: 0;
	font-size: 25px;
	font-weight: 200;
	background-color: rgba(0, 0, 0, 0.5);
`

export const ProductName = styled.h1`
	font-size: 45px;
	font-weight: 600;
	text-align: center;
	margin: 0;
`

export const ProductYear = styled.h2`
	font-size: 22px;
	font-weight: 300;
	margin: 0;
	text-align: right;
`
export const ProductSpan = styled.span<{ mounted: boolean }>`
	margin: 0 auto;
	opacity: ${({ mounted }) => (mounted ? '1' : '0')};
	transition: 0.3s all;
`
