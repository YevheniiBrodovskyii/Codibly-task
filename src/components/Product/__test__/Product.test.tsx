import Product from '../Product'

import { IProduct } from '../product.interface'

import { RenderResult } from '@testing-library/react'

import { renderWithProviders } from '../../../utils/test-utils'

describe('Main tests', () => {
	let component: RenderResult
	const props: IProduct = {
		id: 1,
		name: 'Test',
		year: 2023,
		color: '#000000',
		pantone_value: '20-2023',
	}
	beforeEach(() => {
		component = renderWithProviders(<Product product={props} />)
	})

	it('should match snapshot on initial render', () => {
		expect(component).toMatchSnapshot()
	})

	// it('should trigger fetch arrow button', async () => {

	// })
})
