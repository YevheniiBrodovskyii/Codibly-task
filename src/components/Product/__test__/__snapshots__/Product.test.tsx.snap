// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main tests should match snapshot on initial render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jrcTuL epmhYW"
        color="#000000"
      >
        <h3
          class="sc-kDvujY bHtcak"
        >
          Item id: 
          1
        </h3>
        <h1
          class="sc-ipEyDJ bKvrlp"
        >
          Test
        </h1>
        <span
          class="sc-eDWCr ikxCTO"
        >
          Click for more...
        </span>
        <h2
          class="sc-csuSiG jaQmxB"
        >
          2023
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jrcTuL epmhYW"
      color="#000000"
    >
      <h3
        class="sc-kDvujY bHtcak"
      >
        Item id: 
        1
      </h3>
      <h1
        class="sc-ipEyDJ bKvrlp"
      >
        Test
      </h1>
      <span
        class="sc-eDWCr ikxCTO"
      >
        Click for more...
      </span>
      <h2
        class="sc-csuSiG jaQmxB"
      >
        2023
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
